### PR TITLE
Add long error explanation for E0708 #61137

### PR DESCRIPTION
**Pull request by bishtpawan**
Add long explanation for the E0708 error code
Part of #61137

r? @GuillaumeGomez

### CODE REVIEW

File: src/librustc_error_codes/error_codes/E0708.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+`async` non-`move` closures with parameters are not currently supported.
```
Comment (Dylan-DPC) [inline src/librustc_error_codes/error_codes/E0708.md]:
```suggestion
`async` non-`move` closures with parameters are currently not supported.
```

File: src/librustc_error_codes/error_codes/E0708.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+#![feature(async_closure)]
+
+fn main() {
+    let add_one = async |num: u8| { // error!
+        num + 1
+    };
+}
+```
+
+`async` with non-move currently not supported with the current
```
Comment (Dylan-DPC) [inline src/librustc_error_codes/error_codes/E0708.md]:
```suggestion
`async` with non-move is currently not supported with the 
```

**Review by Dylan-DPC-zz — CHANGES_REQUESTED** [review]

File: src/librustc_error_codes/error_codes/E0708.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
+`async` non-`move` closures with parameters are currently not supported.
+
+Erroneous code example:
+
+```compile_fail,E0708
```
Comment (GuillaumeGomez) [inline src/librustc_error_codes/error_codes/E0708.md:5]:
````suggestion
```compile_fail,E0708,edition2018
````

File: src/librustc_error_codes/error_codes/E0708.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    let add_one = async |num: u8| { // error!
+        num + 1
+    };
+}
+```
+
+`async` with non-move is currently not supported with the current
+version, you can use successfully by using move:
+
+```
```
Comment (GuillaumeGomez) [inline src/librustc_error_codes/error_codes/E0708.md:18]:
````suggestion
```edition2018
````

**Comment by GuillaumeGomez** [comment]
Now please squash your commits then r=me. :)

**Comment by bishtpawan** [comment]
r= @GuillaumeGomez

**Comment by GuillaumeGomez** [comment]
Thanks!

@bors: r+ rollup

**Pull request merged**